cmd/vicadmin/vicadm.go

```diff
@@ -529,9 +547,22 @@ func (s *server) listenPort() int {
 	return s.l.Addr().(*net.TCPAddr).Port
 }
 
+func basicHTTPAuthenticator(handler http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		user, password, ok := r.BasicAuth()
+		if !ok || (user != "root" || password != "thisisinsecure") {
+			w.Header().Add("WWW-Authenticate", "Basic")
+			w.Header().Add("Realm", "vicadmin")
```

Comment:
Realm is not its own header, but a parameter of the WWW-Authenticate header.  See https://tools.ietf.org/html/rfc2617#section-2